presences/src/main/java/fr/openent/presences/service/impl/DefaultAlertService.java

```diff
@@ -258,4 +249,54 @@ public Future<JsonObject> resetStudentAlertsCount(String structureId, String stu
 
         return promise.future();
     }
+
+
+    public Future<Integer> getPageCount(AlertFilterModel alertFilter){
+        Promise<Integer> promise = Promise.promise();
+        JsonArray params = new JsonArray();
+
+        String query = " SELECT count(distinct(student_id, type)) as count from "
+                + Presences.dbSchema +  ".alerts ";
+
+        query += alertsFiltersQuery(alertFilter, params);
+
+        Sql.getInstance().prepared(query,params,SqlResult.validUniqueResultHandler(res -> {
+            if (res.isLeft()) {
+                promise.fail(res.left().getValue());
+            } else {
+                promise.complete(res.right().getValue().getInteger(Field.COUNT, 0));
+            }
+        }));
+        return promise.future();
+    }
+
+    private String alertsFiltersQuery(AlertFilterModel alertFilter, JsonArray params) {
+        StringBuilder conditions = new StringBuilder();
+        if (alertFilter.getStructureId() != null) {
+            conditions.append(" AND");
+            conditions.append(SQLHelper.structureId(alertFilter.getStructureId(), params));
+        }
+        if (!alertFilter.getTypes().isEmpty()) {
+            params.addAll(new JsonArray(alertFilter.getTypes()));
+            conditions.append(" AND");
+            conditions.append(" type IN ").append(Sql.listPrepared(alertFilter.getTypes()));
+            conditions.append(SQLHelper.filterList("type", alertFilter.getTypes(), params));
```

Comment:
donc ta requête ça va être "... AND type in (?, ?, ?) AND type in (?, ?, ?)" ?
tu es sûr de vouloir de vouloir checker 2 fois le type ?